rs/rosetta-api/client/src/lib.rs

```diff
@@ -98,4 +211,268 @@ impl RosettaClient {
+                            signer_keypair.get_pb_key()
+                        )
+                    })?;
+                    if !verification_key
+                        .verify_signature(signable_bytes.as_slice(), signed_bytes.as_slice())
+                    {
+                        bail!("Signature verification failed")
+                    };
+                }
+                _ => bail!(
+                    "Unsupported curve type: {:?}",
+                    signer_keypair.get_curve_type()
+                ),
+            }
+
+            signatures.push(signature);
+        }
+
+        Ok(signatures)
+    }
+
+    pub async fn make_submit_and_wait_for_transaction<T: RosettaSupportedKeyPair>(
+        &self,
+        signer_keypair: &T,
+        network_identifier: NetworkIdentifier,
+        operations: Vec<Operation>,
+        memo: Option<u64>,
+        created_at_time: Option<u64>,
+    ) -> anyhow::Result<ConstructionSubmitResponse> {
+        println!("Making payloads request");
+        println!("Operations: {:?}", operations);
+        let payloads_response = self
+            .construction_payloads(
+                network_identifier.clone(),
+                operations,
+                Some(vec![signer_keypair.into()]),
+                Some(ConstructionPayloadsRequestMetadata {
+                    memo,
+                    created_at_time,
+                    ingress_end: None,
+                    ingress_start: None,
+                }),
+            )
+            .await?;
+        println!("Signing transaction");
+        let signatures = Self::sign_transaction(signer_keypair, payloads_response.clone())?;
+        println!("Combining transaction");
+        let combine_response = self
+            .construction_combine(
+                network_identifier.clone(),
+                payloads_response.unsigned_transaction,
+                signatures,
+            )
+            .await?;
+        println!("Submitting transaction");
+        let submit_response = self
+            .construction_submit(
+                network_identifier.clone(),
+                combine_response.signed_transaction,
+            )
+            .await?;
+
+        // We need to wait for the transaction to be added to the blockchain
+        let mut tries = 0;
+        while tries < 10 {
+            let transaction = self
+                .search_transactions(
+                    &SearchTransactionsRequest::builder(network_identifier.clone())
+                        .with_transaction_identifier(submit_response.transaction_identifier.clone())
+                        .build(),
+                )
+                .await?;
+            if !transaction.transactions.is_empty() {
+                return Ok(submit_response);
+            }
+
+            tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+            tries += 1;
+        }
+
+        bail!("Transaction was not added to the blockchain after 10 seconds")
```

Comment:
nit: Would it make sense to take the number of seconds from `tries`, so that in case someone modifies that value, this message would still be accurate?